metricscaler sample: 'test_scale_bigtable' fails in setup with NotFound

From this [Kokoro build](https://source.cloud.google.com/results/invocations/89e0bb49-0df6-47d9-a9b4-bfd8790f6e8a/targets/github%2Fpython-bigtable%2Fsamples%2Fmetricscaler/tests):

```
    def test_scale_bigtable(instance):
        bigtable_client = bigtable.Client(admin=True)

        instance = bigtable_client.instance(BIGTABLE_INSTANCE)
>       instance.reload()

metricscaler_test.py:109:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../google/cloud/bigtable/instance.py:364: in reload
    instance_pb = self._client.instance_admin_client.get_instance(
../../google/cloud/bigtable_admin_v2/services/bigtable_instance_admin/client.py:605: in get_instance
    response = rpc(request, retry=retry, timeout=timeout, metadata=metadata,)
.nox/py-3-8/lib/python3.8/site-packages/google/api_core/gapic_v1/method.py:145: in __call__
    return wrapped_func(*args, **kwargs)
.nox/py-3-8/lib/python3.8/site-packages/google/api_core/retry.py:285: in retry_wrapped_func
    return retry_target(
.nox/py-3-8/lib/python3.8/site-packages/google/api_core/retry.py:188: in retry_target
    return target()
.nox/py-3-8/lib/python3.8/site-packages/google/api_core/grpc_helpers.py:69: in error_remapped_callable
    six.raise_from(exceptions.from_grpc_error(exc), exc)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

value = None
from_value = <_InactiveRpcError of RPC that terminated with:
	status = StatusCode.NOT_FOUND
	details = "Instance projects/python-do...ge":"Instance projects/python-docs-samples-tests/instances/metric-scale-test-a60739a9-3 not found.","grpc_status":5}"
>

>   ???
E   google.api_core.exceptions.NotFound: 404 Instance projects/python-docs-samples-tests/instances/metric-scale-test-a60739a9-3 not found.

<string>:3: NotFound
```
Likely just needs an EC check.

--- a/samples/metricscaler/metricscaler_test.py
+++ b/samples/metricscaler/metricscaler_test.py
@@ -141,19 +141,21 @@ def test_scale_bigtable(instance):
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, True)
 
-    expected_predicate = ClusterNodeCountPredicate(
+    scaled_node_count_predicate = ClusterNodeCountPredicate(
         original_node_count + SIZE_CHANGE_STEP
     )
+    scaled_node_count_predicate.__name__ = "scaled_node_count_predicate"
     _scaled_node_count = RetryInstanceState(
-        instance_predicate=expected_predicate, max_tries=10,
+        instance_predicate=scaled_node_count_predicate, max_tries=10,
     )
     _scaled_node_count(cluster.reload)()
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, False)
 
-    restored_predicate = ClusterNodeCountPredicate(original_node_count)
+    restored_node_count_predicate = ClusterNodeCountPredicate(original_node_count)
+    restored_node_count_predicate.__name__ = "restored_node_count_predicate"
     _restored_node_count = RetryInstanceState(
-        instance_predicate=restored_predicate, max_tries=10,
+        instance_predicate=restored_node_count_predicate, max_tries=10,
     )
     _restored_node_count(cluster.reload)()
 

--- a/samples/metricscaler/metricscaler_test.py
+++ b/samples/metricscaler/metricscaler_test.py
@@ -23,6 +23,7 @@ from mock import Mock, patch
 
 import pytest
 from test_utils.retry import RetryInstanceState
+from test_utils.retry import RetryResult
 
 from metricscaler import get_cpu_load
 from metricscaler import get_storage_utilization
@@ -74,6 +75,10 @@ def instance():
                                    default_storage_type=storage_type)
         instance.create(clusters=[cluster])
 
+        # Eventual consistency check
+        retry_found = RetryResult(bool)
+        retry_found(instance.exists)()
+
     yield
 
     instance.delete()
@@ -96,6 +101,10 @@ def dev_instance():
         cluster = instance.cluster(cluster_id, location_id=BIGTABLE_ZONE,
                                    default_storage_type=storage_type)
         instance.create(clusters=[cluster])
+
+        # Eventual consistency check
+        retry_found = RetryResult(bool)
+        retry_found(instance.exists)()
 
     yield
 

--- a/samples/metricscaler/metricscaler_test.py
+++ b/samples/metricscaler/metricscaler_test.py
@@ -111,6 +111,18 @@ def dev_instance():
     instance.delete()
 
 
+class ClusterNodeCountPredicate:
+    def __init__(self, expected_node_count):
+        self.expected_node_count = expected_node_count
+
+    def __call__(self, cluster):
+        expected = self.expected_node_count
+        print(
+            f"Expected node count: {expected}; found: {cluster.serve_nodes}"
+        )
+        return cluster.serve_nodes == expected
+
+
 def test_scale_bigtable(instance):
     bigtable_client = bigtable.Client(admin=True)
 
@@ -129,18 +141,19 @@ def test_scale_bigtable(instance):
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, True)
 
-    expected_count = original_node_count + SIZE_CHANGE_STEP
+    expected_predicate = ClusterNodeCountPredicate(
+        original_node_count + SIZE_CHANGE_STEP
+    )
     _scaled_node_count = RetryInstanceState(
-        instance_predicate=lambda c: c.serve_nodes == expected_count,
-        max_tries=10,
+        instance_predicate=expected_predicate, max_tries=10,
     )
     _scaled_node_count(cluster.reload)()
 
     scale_bigtable(BIGTABLE_INSTANCE, BIGTABLE_INSTANCE, False)
 
+    restored_predicate = ClusterNodeCountPredicate(original_node_count)
     _restored_node_count = RetryInstanceState(
-        instance_predicate=lambda c: c.serve_nodes == original_node_count,
-        max_tries=10,
+        instance_predicate=restored_predicate, max_tries=10,
     )
     _restored_node_count(cluster.reload)()
 

